# **ASSIGNMENT**

**Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose
you have collected data on the amount of time students spend studying for an exam and their final exam
scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.**

In [1]:
import pandas as pd
data_set=pd.DataFrame(data={"No. of hours studied":[10,2,5,7,6,12,13],"Scores in Exam":[80,30,50,70,60,90,98]})
data_set

No. of hours studied  Scores in Exam
0                    10              80
1                     2              30
2                     5              50
3                     7              70
4                     6              60
5                    12              90
6                    13              98

In [2]:
data_set.corr(method="pearson")

No. of hours studied  Scores in Exam
No. of hours studied              1.000000        0.988395
Scores in Exam                    0.988395        1.000000

Interpreting the result, we can see that the Pearson correlation coefficient between the number of hours studied and the scores in the exam is 0.988395, which indicates a strong positive linear relationship between these two variables. This suggests that as the number of hours studied increases, the scores in the exam also tend to increase.

**Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables.
Suppose you have collected data on the amount of sleep individuals get each night and their overall job
satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two
variables and interpret the result.**

In [3]:
dataset_job=pd.DataFrame({"Amount of Sleep (hours)":[6.5,7.2,5.8,8.1,6.9,7.5,5.5,8.3,7.8],
                          "Job Satisfaction (Scale 1-10)":[7,6,5,8,6,7,4,9,8]})            
dataset_job

Amount of Sleep (hours)  Job Satisfaction (Scale 1-10)
0                      6.5                              7
1                      7.2                              6
2                      5.8                              5
3                      8.1                              8
4                      6.9                              6
5                      7.5                              7
6                      5.5                              4
7                      8.3                              9
8                      7.8                              8

In [4]:
dataset_job.corr(method="spearman")

Amount of Sleep (hours)  \
Amount of Sleep (hours)                       1.000000   
Job Satisfaction (Scale 1-10)                 0.919905   

                               Job Satisfaction (Scale 1-10)  
Amount of Sleep (hours)                             0.919905  
Job Satisfaction (Scale 1-10)                       1.000000

Interpreting the result, we can see that the Spearman's rank correlation coefficient between the amount of sleep individuals get each night and their overall job satisfaction level is 0.919905. This indicates a moderately strong positive monotonic relationship between these two variables. Therefore, as the amount of sleep increases, the job satisfaction level tends to increase as well. It is important to note that Spearman's rank correlation measures the monotonic relationship and not the linearity of the data.

**Q3. Suppose you are conducting a study to examine the relationship between the number of hours of
exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables
for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation
between these two variables and compare the results.**

In [5]:
import pandas as pd

data = {
    'Exercise (hours per week)': [4.5, 3.2, 6.8, 2.1, 5.5, 1.7, 3.9, 2.3, 7.2, 4.8,
                                  3.5, 6.1, 2.9, 5.3, 1.5, 4.2, 2.7, 6.5, 3.8, 7.4,
                                  4.1, 3.0, 5.1, 1.9, 4.9, 2.5, 6.9, 3.7, 7.8, 4.4,
                                  3.3, 6.7, 2.0, 5.6, 1.8, 4.0, 2.8, 7.1, 3.6, 5.2,
                                  1.6, 4.3, 2.6, 6.3, 3.4, 5.9, 2.4, 7.3, 4.7,4.9],
    'BMI': [23.5, 25.1, 27.3, 30.9, 22.8, 29.4, 26.5, 31.2, 25.3, 23.9,
            27.8, 24.6, 29.1, 22.2, 30.6, 26.9, 30.3, 24.8, 27.1, 25.7,
            23.4, 29.7, 22.9, 31.5, 25.5, 27.6, 24.1, 28.9, 23.1, 26.7,
            30.0, 25.1, 28.3, 23.2, 31.0, 26.3, 30.5, 24.9, 27.2, 22.5,
            29.2, 24.7, 28.8, 23.3, 30.2, 25.6, 31.3, 26.0,27.0,28.0],
}

dataset_body = pd.DataFrame(data)
dataset_body.head()

Exercise (hours per week)   BMI
0                        4.5  23.5
1                        3.2  25.1
2                        6.8  27.3
3                        2.1  30.9
4                        5.5  22.8

In [6]:
dataset_body.corr(method="pearson")

Exercise (hours per week)       BMI
Exercise (hours per week)                   1.000000 -0.749178
BMI                                        -0.749178  1.000000

In [7]:
dataset_body.corr(method="spearman")

Exercise (hours per week)       BMI
Exercise (hours per week)                   1.000000 -0.777372
BMI                                        -0.777372  1.000000

Certainly! Let's compare the results of the Pearson correlation coefficient and Spearman's rank correlation for the relationship between the number of hours of exercise per week and body mass index (BMI) in the given dataset.

Assuming you have already calculated the correlations, let's say:

Pearson correlation coefficient: -0.749178
Spearman's rank correlation: -0.777372

Based on these results:

1. Pearson correlation coefficient: The value of -0.749178 suggests a moderately strong negative linear relationship between the number of hours of exercise per week and BMI. It indicates that as the number of hours of exercise increases, the BMI tends to decrease. However, since the value is less than 1, the relationship is not perfectly linear.

2. Spearman's rank correlation: The value of -0.777372 suggests a moderately strong negative monotonic relationship between the variables. It indicates that as the number of hours of exercise increases, there is a tendency for the BMI to decrease. The Spearman's rank correlation captures the overall trend between the variables, regardless of the exact shape of the relationship.

Comparing the results:

Both the Pearson correlation coefficient and Spearman's rank correlation indicate a negative relationship between the number of hours of exercise per week and BMI. However, the Pearson correlation coefficient (-0.749178) reflects the linear relationship, while the Spearman's rank correlation -0.777372 captures the general monotonic trend between the variables.

In conclusion, the analysis suggests that there is a negative relationship between the number of hours of exercise per week and BMI in the given dataset. The strength of the relationship is moderately strong according to both the Pearson correlation coefficient and Spearman's rank correlation.

**Q4. A researcher is interested in examining the relationship between the number of hours individuals
spend watching television per day and their level of physical activity. The researcher collected data on
both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between
these two variables.**

In [8]:
import pandas as pd

data = {
    'TV Hours per Day': [2.5, 1.8, 3.2, 4.0, 2.1, 3.5, 1.2, 2.9, 3.8, 2.7,
                         1.5, 2.3, 4.5, 1.7, 3.1, 2.6, 1.9, 3.7, 2.4, 1.6,
                         2.8, 3.4, 1.3, 2.2, 4.1, 2.0, 3.6, 1.4, 3.3, 2.5,
                         4.2, 1.1, 2.7, 3.9, 2.3, 3.8, 1.5, 2.9, 4.0, 2.1,
                         3.5, 1.2, 2.6, 3.4, 2.0, 3.6, 1.3, 2.2, 3.9, 2.4],
    'Physical Activity Level': [4, 6, 2, 1, 5, 3, 7, 4, 2, 5,
                                6, 4, 1, 5, 3, 6, 7, 2, 4, 6,
                                3, 5, 7, 4, 2, 5, 3, 6, 1, 4,
                                2, 5, 6, 3, 4, 2, 7, 5, 3, 1,
                                4, 6, 2, 3, 5, 1, 4, 2, 5, 3],
}

dataset_researcher = pd.DataFrame(data)
dataset_researcher.head()

TV Hours per Day  Physical Activity Level
0               2.5                        4
1               1.8                        6
2               3.2                        2
3               4.0                        1
4               2.1                        5

In [9]:
dataset_researcher.corr(method="pearson")

TV Hours per Day  Physical Activity Level
TV Hours per Day                 1.000000                -0.704162
Physical Activity Level         -0.704162                 1.000000

**Q5 is incomplete in Assignment**

**Q6. A company is interested in examining the relationship between the number of sales calls made per day
and the number of sales made per week. The company collected data on both variables from a sample of
30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.**

In [10]:
import pandas as pd

data = {
    'Sales Calls per Day': [25, 30, 20, 35, 28, 22, 27, 31, 24, 29,
                            26, 23, 32, 21, 27, 30, 25, 34, 22, 29,
                            28, 31, 19, 26, 33, 24, 29, 23, 27, 30],
    'Sales per Week': [12, 15, 9, 18, 14, 10, 13, 16, 11, 15,
                       12, 10, 17, 9, 13, 15, 12, 19, 10, 14,
                       15, 16, 8, 11, 18, 10, 15, 9, 13, 15],
}

dataset_sales = pd.DataFrame(data)
dataset_sales.head()

Sales Calls per Day  Sales per Week
0                   25              12
1                   30              15
2                   20               9
3                   35              18
4                   28              14

In [11]:
dataset_sales.corr(method="pearson")

Sales Calls per Day  Sales per Week
Sales Calls per Day             1.000000        0.979522
Sales per Week                  0.979522        1.000000

--------------------------------